zerocopy-derive/src/lib.rs

```diff
@@ -87,11 +76,30 @@ macro_rules! derive {
         #[proc_macro_derive($trait)]
         pub fn $outer(ts: proc_macro::TokenStream) -> proc_macro::TokenStream {
             let ast = syn::parse_macro_input!(ts as DeriveInput);
-            $inner(&ast).into()
+            $inner(&ast).into_ts().into()
         }
     };
 }
 
+trait IntoTokenStream {
```

Comment:
Maybe. It's just added boilerplate in a number of places. This allows the support to be more local rather than force all of the functions to return `Result`s and wrap things in `Ok`, but either's fine - it's not a huge difference either way.